Source/WebKit/WebProcess/ApplePay/WebPaymentCoordinator.cpp

```diff
@@ -78,11 +78,16 @@ std::optional<String> WebPaymentCoordinator::validatedPaymentNetwork(const Strin
 
 bool WebPaymentCoordinator::canMakePayments()
 {
-    bool canMakePayments;
-    if (!sendSync(Messages::WebPaymentCoordinatorProxy::CanMakePayments(), Messages::WebPaymentCoordinatorProxy::CanMakePayments::Reply(canMakePayments)))
-        return false;
-
-    return canMakePayments;
+    auto now = MonotonicTime::now();
+    if (now - m_timestampOfLastCanMakePaymentsRequest > 1_min || !m_lastCanMakePaymentsResult) {
+        bool canMakePayments;
+        if (!sendSync(Messages::WebPaymentCoordinatorProxy::CanMakePayments(), Messages::WebPaymentCoordinatorProxy::CanMakePayments::Reply(canMakePayments)))
```

Comment:
The "right" design for this is probably caching indefinitely (and maybe even ahead of time, instead of synchronously?) and invalidating on a notification -- do you know if that exists? This is fine for now, anyway.

Reply:
yeah that definitely would be the most ideal.

the PassKit folks tell me that there is a notification that's dispatched whenever the pass library changes (not necessarily specific to Apple Pay availability/capability), but there are additional complexities on macOS that mean that  that is not the only determinant (and the others are not broadcast as notifications).

i'll land this as-is for now and file a followup to look into optimizing